src/plugins/intel_cpu/src/cpu_memory.cpp

```diff
@@ -34,10 +34,26 @@ namespace {
     }
 }   // namespace
 
-Memory::Memory(const dnnl::engine& eng) :
-    eng(eng), mgrHandle(std::make_shared<DnnlMemoryMngr>(std::unique_ptr<MemoryMngrWithReuse>(new MemoryMngrWithReuse())), this), dnnlMemHandle(this) {}
-Memory::Memory(const dnnl::engine& eng, std::unique_ptr<IMemoryMngr> mngr) :
-    eng(eng), mgrHandle(std::make_shared<DnnlMemoryMngr>(std::move(mngr)), this), dnnlMemHandle(this) {}
+Memory::Memory(const dnnl::engine& eng, MemoryDescPtr desc, const void* data, bool pads_zeroing) :
+    pMemDesc(desc),
+    eng(eng),
+    mgrHandle(std::make_shared<DnnlMemoryMngr>(std::unique_ptr<MemoryMngrWithReuse>(new MemoryMngrWithReuse())), this),
+    dnnlMemHandle(this) {
+        Create(pMemDesc, data, pads_zeroing);
+    }
+
+Memory::Memory(const dnnl::engine& eng, const MemoryDesc& desc, const void* data, bool pads_zeroing) :
+    Memory::Memory(eng, desc.clone(), data, pads_zeroing) {}
+
+Memory::Memory(const dnnl::engine& eng, MemoryDescPtr desc, MemoryMngrPtr mngr) :
+    pMemDesc(desc), eng(eng), mgrHandle(mngr, this), dnnlMemHandle(this) {
+        bool memAllocated = mgrHandle->getRawPtr();
```

Comment:
Is it an expected return type bool?

Reply:
This is a null check actually. So any address except null is converted to true.